haproxy_exporter.go

```diff
@@ -358,13 +359,31 @@ func main() {
 		haProxyScrapeUri          = flag.String("haproxy.scrape_uri", "http://localhost/;csv", "URI on which to scrape HAProxy.")
 		haProxyServerMetricFields = flag.String("haproxy.server_metric_fields", "", "If specified, only export the given csv fields. Comma-seperated list of numbers. See http://cbonte.github.io/haproxy-dconv/configuration-1.5.html#9.1")
 		haProxyTimeout            = flag.Duration("haproxy.timeout", 5*time.Second, "Timeout for trying to get stats from HAProxy.")
-		_                         = flag.Duration("haproxy.scrape_interval", 0, "DEPRECATED. Not used anymore.")
+		haProxyPidFile            = flag.String("haproxy.pid_file", "", "Path to haproxy's pid file.")
 	)
 	flag.Parse()
 
 	exporter := NewExporter(*haProxyScrapeUri, *haProxyServerMetricFields, *haProxyTimeout)
 	prometheus.MustRegister(exporter)
 
+	if *haProxyPidFile != "" {
+		procExporter := prometheus.NewProcessCollectorPIDFn(
+			func() int {
```

Comment:
@grobie what's the reasoning why the pid function wouldn't return `(int, error)`? Now when there's an error here, it ends up doing `NewProc(0)`...